// travel-ui/src/dialog/modal.js
/**
 * Created by max on 05.08.16.
 */
import block from 'bem-cn';
import React, {Component, PropTypes} from 'react';
import Dialog from './dialog'

let b = block('TravelUiModal');

class Modal extends Component {
    
    static displayName = 'Modal';
    
    static contextTypes = {
        componentStyle: React.PropTypes.object
    };
    
    static propTypes = {
        className: PropTypes.string,
        isShow   : PropTypes.bool,
        small    : PropTypes.bool,
        medium   : PropTypes.bool,
        large    : PropTypes.bool
    };
    
    static defaultProps = {
        className: '',
        isShow   : false,
        small    : true,
        medium   : false,
        large    : false,
    };
    
    render() {
        return (
            <Dialog isShow={this.props.isShow}
                    isOverlay={this.props.isOverlay}>
                <div className={b({small: this.props.small, medium: this.props.medium, large: this.props.large})}>
                    {this.props.children}
                </div>
            </Dialog>
        )
    }
}

export default Modal
