// travel-ui/example/example.js
/**
 * Created by max on 03.08.16.
 */
import React, {Component} from 'react'
import {connect} from 'react-redux'

// import ExampleForm from './form'
import {Modal} from '../src/index'
// let Dialog = TravelUi.Dialog
// let Modal = TravelUi.Modal


class Example extends Component {
    
    constructor(props) {
        super(props)
        
        this.state = {
            showDialog     : false,
            showModalSmall: false,
            showModalMedium: false,
            showModalLarge: false,
        };
    }
    
    handleSubmit() {
        
    }
    
    renderDialog() {
        this.setState({
            showDialog: !this.state.showDialog
        })
    }
    
    renderModalSmall() {
        this.setState({
            showModalSmall: !this.state.showModalSmall
        })
    }
    
    renderModalMedium() {
        this.setState({
            showModalMedium: !this.state.showModalMedium
        })
    }
    
    renderModalLarge() {
        this.setState({
            showModalLarge: !this.state.showModalLarge
        })
    }
    
    render() {
        return (
            <div>
                
                <button onClick={this.renderDialog.bind(this)}>render dialog</button>
                <button onClick={this.renderModalSmall.bind(this)}>renderModalSmall</button>
                
                
                <Modal isShow={this.state.showModalSmall}
                       isOverlay={this.renderModalSmall.bind(this)}
                       small={true}>
                    child dialog
                    <br/>
                    child dialog
                    <br/>
                    child dialog
                    <br/>
                    child dialog
                    <br/>
                    child dialog
                    <br/>
                    child dialog
                    <br/>
                    child dialog
                    <br/>
                    <button onClick={this.renderModalMedium.bind(this)}>renderModalMedium</button>
                    <button onClick={this.renderModalLarge.bind(this)}>renderModalLarge</button>
                </Modal>
                <Modal isShow={this.state.showModalMedium}
                       isOverlay={this.renderModalMedium.bind(this)}
                       medium={true}>
                    child dialog
                    <br/>
                    child dialog
                    <br/>
                    child dialog
                    <br/>
                    child dialog
                    <br/>
                    child dialog
                    <br/>
                    <button onClick={this.renderModalSmall.bind(this)}>renderModalSmall</button>
                    <button onClick={this.renderModalLarge.bind(this)}>renderModalLarge</button>
                </Modal>
                <Modal isShow={this.state.showModalLarge}
                       isOverlay={this.renderModalLarge.bind(this)}
                       large={true}>
                    child dialog
                    <br/>
                    child dialog
                    <br/>
                    child dialog
                    <br/>
                    child dialog
                    <br/>
                    <button onClick={this.renderModalLarge.bind(this)}>renderModalLarge</button>
                    <button onClick={this.renderModalMedium.bind(this)}>renderModalMedium</button>
                    <button onClick={this.renderModalSmall.bind(this)}>renderModalSmall</button>
                </Modal>
            </div>
        )
    }
}

function mapStateToProps(state) {
    return {
        user: state.user
    }
}

export default connect(mapStateToProps)(Example)
